### PR TITLE
Add support for rust 1.30.0

### PR DESCRIPTION
**Pull request by iohe**
My first PR, be gentle.
Adding rust 1.30.0 to meta-rust

### CODE REVIEW

File: recipes-devtools/cargo/cargo-0.30.0.inc [hunk 1]
```diff
@@ -0,0 +1,13 @@
+CARGO_SNAPSHOT = "cargo-0.29.0-${BUILD_ARCH}-unknown-linux-gnu"
```
Comment (otavio) [inline recipes-devtools/cargo/cargo-0.30.0.inc]: Please use 0.30.0 here as well.

File: recipes-devtools/cargo/cargo-0.30.0.inc [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+SRC_URI[cargo-snapshot-x86_64.md5sum] = "f9d283378b9b0d8457f737c2a6bfbc67"
+SRC_URI[cargo-snapshot-x86_64.sha256sum] = "a5e7749767c47669ed9b6e32c6fb8eda6b0fe1c63ac73a6d4666b5c1352bad24"
+
```
Comment (otavio) [inline recipes-devtools/cargo/cargo-0.30.0.inc]: Those will change.

File: recipes-devtools/cargo/cargo-0.30.0.inc (hunk 1 above)
Comment (ashie) [reply recipes-devtools/cargo/cargo-0.30.0.inc]: Probably using the previous version here is intentional.
It's for building rust/cargo themselves (stage0).

* https://github.com/rust-lang/rust/blob/master/src/stage0.txt
* https://github.com/rust-lang/rust/blob/master/src/bootstrap/README.md

```
# Currently Rust always bootstraps from the previous stable release, and in our
# train model this means that the master branch bootstraps from beta, beta
# bootstraps from current stable, and stable bootstraps from the previous stable
# release.
```

Comment (otavio) [reply recipes-devtools/cargo/cargo-0.30.0.inc]: Agreed; I didn't see it.

Thanks for the reply on top of mine :+1:

File: recipes-devtools/rust/rust-source-1.30.0.inc [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Specifics for Rust 1.30.0
+
+# SRC_URI += "https://static.rust-lang.org/dist/rustc-${PV}-src.tar.gz;name=rust"
+SRC_URI += " \
+        https://static.rust-lang.org/dist/rustc-${PV}-src.tar.gz;name=rust \
+        file://skip-crate-build-id-check.patch;patchdir=${WORKDIR}/rustc-${PV}-src \
```
Comment (otavio) [inline recipes-devtools/rust/rust-source-1.30.0.inc]: This patch file has not been committed, please update the PR to include it or drop it.

**Comment by srwalter** [comment]
We're past version 1.30 already

**Pull request closed without merging**